samcli/commands/local/lib/local_lambda.py

```diff
@@ -141,6 +142,12 @@ def _make_env_vars(self, function):
         #
         # Standard format is {FunctionName: {key:value}, FunctionName: {key:value}}
         # CloudFormation parameter file is {"Parameters": {key:value}}
+
+        for fn in self.env_vars_values:
+            if not isinstance(self.env_vars_values.get(fn), dict):
```

Comment:
This can be simplified further. You only care about the values of the dictionary so you can iterate over the values of the dictionary and check types on that.

```
for env_var_value in self.env_vars_values.values():
    if not isinstance(env_var_value, dict):
        raise
```

You can also loop over keys and values at the same time by if you really need/want the key for something:

```
for key, value in self.env_vars_values.items():
    pass
```